src/variant.rs

```diff
@@ -0,0 +1,356 @@
+// Copyright 2013-2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+//! `Variant` binding and helper traits.
+//!
+//! [`Variant`](struct.Variant.html) is an immutable dynamically-typed generic
+//! container. Its type and value are defined at construction and never change.
+//!
+//! `Variant` types are described by [`VariantType`](../struct.VariantType.html)
+//! "type strings".
+//!
+//! Although `GVariant` supports arbitrarily complex types, currently this
```

Comment:
> Although `GVariant` supports arbitrarily complex types, currently this binding is limited to the basic ones:

Instead I propose: "Although `GVariant` supports arbitrarily complex types, this binding is currently limited to the basic ones:"